dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/config/ServiceConfig.java

```diff
@@ -153,203 +203,220 @@ public RetryTemplate retryTemplate()
         return retryTemplate;
     }
 
-    private final static List<Class<? extends ValidationCheck>> CREATE_UPDATE_CHECKS = newArrayList(
-        DuplicateIdsCheck.class,
-        ValidationHooksCheck.class,
-        SecurityCheck.class,
-        SchemaCheck.class,
-        UniquenessCheck.class,
-        UniqueMultiPropertiesCheck.class,
-        MandatoryAttributesCheck.class,
-        UniqueAttributesCheck.class,
-        ReferencesCheck.class,
-        NotOwnerReferencesCheck.class );
-
-    private final static List<Class<? extends ValidationCheck>> CREATE_CHECKS = newArrayList(
-        DuplicateIdsCheck.class,
-        ValidationHooksCheck.class,
-        SecurityCheck.class,
-        CreationCheck.class,
-        SchemaCheck.class,
-        UniquenessCheck.class,
-        UniqueMultiPropertiesCheck.class,
-        MandatoryAttributesCheck.class,
-        UniqueAttributesCheck.class,
-        ReferencesCheck.class,
-        NotOwnerReferencesCheck.class );
-
-    private final static List<Class<? extends ValidationCheck>> UPDATE_CHECKS = newArrayList(
-        DuplicateIdsCheck.class,
-        ValidationHooksCheck.class,
-        SecurityCheck.class,
-        UpdateCheck.class,
-        SchemaCheck.class,
-        UniquenessCheck.class,
-        UniqueMultiPropertiesCheck.class,
-        MandatoryAttributesCheck.class,
-        UniqueAttributesCheck.class,
-        ReferencesCheck.class,
-        NotOwnerReferencesCheck.class );
-
-    private final static List<Class<? extends ValidationCheck>> DELETE_CHECKS = newArrayList(
-        SecurityCheck.class,
-        DeletionCheck.class );
-
     @Bean( "validatorMap" )
-    public Map<ImportStrategy, List<Class<? extends ValidationCheck>>> validatorMap()
+    public Map<ImportStrategy, List<ValidationCheck>> validatorMap()
     {
         return ImmutableMap.of(
-            ImportStrategy.CREATE_AND_UPDATE, CREATE_UPDATE_CHECKS,
-            CREATE, CREATE_CHECKS,
-            ImportStrategy.UPDATE, UPDATE_CHECKS,
-            ImportStrategy.DELETE, DELETE_CHECKS );
+            CREATE_AND_UPDATE, newArrayList(
```

Comment:
If performance is a reason wouldn't it make more sense to do this once - wrap list and map in unmodifiable ones and keep that as internal state that can be returned directly?

Reply:
I prefer to have them listed explicitly since potentially there might exist a Check which only makes sense for a given ImportStrategy. That's why they are listed. Moreover, this code only makes sense during spring bean configuration to create beans.